PWGHF/D2H/TableProducer/candidateCreatorCharmResoReduced.cxx

```diff
@@ -242,60 +251,87 @@ struct HfCandidateCreatorCharmResoReduced {
       std::array<float, 3> pVecD = {candD.px(), candD.py(), candD.pz()};
       std::array<int, 3> dDaughtersIds = {candD.prong0Id(), candD.prong1Id(), candD.prong2Id()};
 
-      // loop on V0 candidates
+      // loop on V0 or track candidates
       bool alreadyCounted{false};
-      for (const auto& candV0 : candsV0) {
+      for (const auto& candV0Tr : candsV0Tr) {
         if (rejectDV0PairsWithCommonDaughter) {
           const std::array<int, 3> dDaughtersIDs = {candD.prong0Id(), candD.prong1Id(), candD.prong2Id()};
-          if (std::find(dDaughtersIDs.begin(), dDaughtersIDs.end(), candV0.prong0Id()) != dDaughtersIDs.end() || std::find(dDaughtersIDs.begin(), dDaughtersIDs.end(), candV0.prong1Id()) != dDaughtersIDs.end()) {
+          if constexpr (channel == DecayChannel::DstarTrack){
+            if(std::find(dDaughtersIDs.begin(), dDaughtersIDs.end(),candV0Tr.globalIndex()) != dDaughtersIDs.end()) {
+               continue;
+            }
+           }else{
+           if(std::find(dDaughtersIDs.begin(), dDaughtersIDs.end(), candV0Tr.prong0Id()) != dDaughtersIDs.end() || std::find(dDaughtersIDs.begin(), dDaughtersIDs.end(), candV0Tr.prong1Id()) != dDaughtersIDs.end()) 
+            {
             continue;
+            }
           }
         }
-
-        if (!isV0Selected<channel>(candV0, candD)) {
+        if constexpr (channel != DecayChannel::DstarTrack){
+        if (!isV0Selected<channel>(candV0Tr, candD)) {
           continue;
         }
         if (activateQA && !alreadyCounted) {
           registry.fill(HIST("hSelections"), 1 + Selections::V0Sel);
           alreadyCounted = true;
         }
+        }
+
         float invMassReso{0.};
         float invMassV0{0.};
-        std::array<float, 3> pVecV0 = {candV0.px(), candV0.py(), candV0.pz()};
+        std::array<float, 3> pVecV0 = {candV0Tr.px(), candV0Tr.py(), candV0Tr.pz()};
         float ptReso = RecoDecay::pt(RecoDecay::sumOfVec(pVecV0, pVecD));
+        
+        if constexpr (channel==DecayChannel::DstarTrack){
+            if (candD.dType() > 0) {
+            invMassReso= RecoDecay::m(std::array{pVecD, pVecV0},std::array{massDstar, massProton});
+            }else{
+            invMassReso= RecoDecay::m(std::array{pVecD, pVecV0},std::array{massDstar, massProton});
+            }
```

Comment:
No need to distinguish the two cases, if the mass is computed using the PDG D* mass
```suggestion
            invMassReso= RecoDecay::m(std::array{pVecD, pVecV0Tr}, std::array{massDstar, massProton});
```